katharsis-core/src/main/java/io/katharsis/core/internal/resource/AnnotationResourceInformationBuilder.java

```diff
@@ -155,32 +151,28 @@ public static String getResourceType(Type genericType, ResourceInformationBuilde
 		}
 
 		for (ResourceFieldWrapper fieldWrapper : resourceGetterFields) {
-			if (!fieldWrapper.isDiscarded()) {
-				String originalName = fieldWrapper.getResourceField().getUnderlyingName();
-				AnnotatedResourceField field = fieldWrapper.getResourceField();
-				if (resourceFieldMap.containsKey(originalName)) {
-					resourceFieldMap.put(originalName, mergeAnnotations(resourceFieldMap.get(originalName), field, context));
-				} else if (!hasDiscardedField(fieldWrapper, resourceClassFields)) {
-					resourceFieldMap.put(originalName, field);
-				}
-			}
-		}
-
-		return discardIgnoredField(resourceFieldMap.values());
-	}
-
-	private List<AnnotatedResourceField> discardIgnoredField(Collection<AnnotatedResourceField> resourceFieldValues) {
-		List<AnnotatedResourceField> resourceFields = new LinkedList<>();
-		for (AnnotatedResourceField resourceField : resourceFieldValues) {
-			if (!resourceField.isAnnotationPresent(JsonIgnore.class)) {
-				resourceFields.add(resourceField);
+			String originalName = fieldWrapper.getResourceField().getUnderlyingName();
+			AnnotatedResourceField field = fieldWrapper.getResourceField();
+			boolean getterIsIgnored = fieldWrapper.isDiscarded();
+			boolean didAddField = resourceFieldMap.containsKey(originalName);
+			boolean didDiscardField = !didAddField && hasField(fieldWrapper, resourceClassFields);
+
+			if (didAddField && getterIsIgnored) {
+				// we added the field, but the getter is ignored, so we discard the field too
+				resourceFieldMap.remove(originalName);
+			} else if (didAddField) {
```

Comment:
This else/if conditional is a bit redundant. You can do:

```
if(didAddField) {
  if(getterIsIgnored) {
   resourceFieldMap.remove(originalName);
}
```

It's redundant to check the predicate in the else clause